src/interfaces/tearoffbase.hpp

```diff
@@ -0,0 +1,86 @@
+#ifndef _SRC_INTERFACES_TEAROFFBASE_HPP_
+#define _SRC_INTERFACES_TEAROFFBASE_HPP_
+
+#include "internal/dnmd_platform.hpp"
+#include <atomic>
+#include <memory>
+
+class ControllingIUnknown;
+
+class TearOffUnknown : public IUnknown
+{
+    friend class ControllingIUnknown;
+private:
+    IUnknown* _pUnkOuter;
+
+protected:
+    STDMETHOD(TryGetInterfaceOnThis)(REFIID riid, void** ppvObject) PURE;
```

Comment:
I don't think we want an `HRESULT` here. I say that only because we aren't using it. I would just make it `bool` and that indicates `S_OK` or `E_NOINTERFACE`, there isn't any other `HRESULT` to return.